src/components/animated-icon-wrapper/AnimatedIconWrapper.tsx

```diff
@@ -0,0 +1,39 @@
+import { FC, PropsWithChildren } from 'react';
+import Transition from 'react-transition-group/Transition';
+
+import {
+	defaultStyle,
+	transitionStyles,
+	useAnimatedIconWrapperStyles,
+} from './useAnimatedIconWrapperStyles';
+
+interface AnimatedIconWrapperProps extends PropsWithChildren {
+	durationMs: number;
+	hasAnimationStarted?: boolean;
```

Comment:
Can we rename this to `startAnimation` or `animate` since the wrapper is the one responsible for the animation?